Tests/MoSoAnalyticsTests/MoSoDiscoverTrackerTests.swift

```diff
@@ -0,0 +1,107 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import MoSoCore
+import XCTest
+@testable import MoSoAnalytics
+
+class MoSoDiscoverTrackerTests: XCTestCase {
+    func testTrackRecommendationOpen() {
+        // Given
+        let user = MoSoUser(username: "example@username.org", identifier: "myNickname")
+        let baseTracker = MockBaseTracker(session: MoSoSession(user: user))
+        let discoverTracker = MoSoDiscoverTracker(baseTracker: baseTracker)
+        baseTracker.engagementExpectation = expectation(description: "open recommendation")
```

Comment:
curious to understand why we need to set an expectation here since tests seem to pass without it.

Reply:
@cyndichin I'm getting an error when trying to run the tests if I disable this expectation

Reply:
oh that's so weird, what error? I disable this as well as the `waitForExpectations` I'm curious to see what it is testing.

Reply:
the expectation tests that the method is called, the other checks add to it, testing that the call did exactly what was supposed to do.